expression/expression.go

```diff
@@ -407,6 +408,18 @@ func NewValuesFunc(ctx sessionctx.Context, offset int, retTp *types.FieldType) *
 	}
 }
 
+// NewIsTrueFunc create a new istrue function.
```

Comment:
nit: Could you add a comment here to explain why this function should be present and what does `castNull` mean?

Reply:
Would it be more elegant to add a new eval type `EBoolean` for internal usage and modify the `CAST` function to support the cast to boolean?